### ISSUE TITLE
Migrate form validations to latest version of joi

### ISSUE THREAD

**Issue by davidrapson** [biglotteryfund/blf-alpha#3352]
We are currently running multiple versions of `joi`. An older version for application forms and a newer one for everything else. Most of the prep work has already been done to allow us to finish this migration. The remaining tasks are:

## Upgrade form validations to use new version

The following pull requests document this process. Parked it due to other priorities but they are a good starting point as they call out the key syntax changes to be aware of.

- https://github.com/biglotteryfund/blf-alpha/pull/3246
- https://github.com/biglotteryfund/blf-alpha/pull/3223

Likely to be some subtle API changes there so needs a good amount of testing, but not a lot of actual code changes to get there.

## Delete supporting code

Once that is done the supporting code relating to migration and the old version can be removed

- Delete `joi-extensions` and rename `joi-extensions-next`
- Remove references to old version of `joi`
